### PR TITLE
[CALCITE-2679] Fix a bug that `any_value` can't be used with other ag…

### PR DESCRIPTION
**Pull request by Functor10**
…g functions
The problem appears in nested buckets.

### CODE REVIEW

File: elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/AggregationTest.java [hunk 1]
```diff
@@ -257,6 +257,27 @@ public void anyValue() {
...
+            .returnsUnordered("EXPR$0=1.0; cat1=a; EXPR$2=g",
+                    "EXPR$0=null; cat1=null; EXPR$2=g",
+                    "EXPR$0=7.0; cat1=b; EXPR$2=h");
+
+    CalciteAssert.that()
+            .with(newConnectionFactory())
+            .query("select any_value(cat2), cat1, max(val1) from view group by cat1")
+            .returnsUnordered("EXPR$0=g; cat1=a; EXPR$2=1.0",
+                    "EXPR$0=g; cat1=null; EXPR$2=null",
+                    "EXPR$0=h; cat1=b; EXPR$2=7.0");
```
Comment (michaelmior) [inline elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/AggregationTest.java]: These should probably be separate tests to make debugging easier if anything breaks them in the future. (I know initially there were already a couple asserts in the same tests, but if meaningful names can be given to each test, that would be great!)

**Comment by michaelmior** [comment]
@asereda-gs Do you mind having a look at this?

**Comment by julianhyde** [comment]
Please change the commit comment to not start with 'fix'. Just provide the description of the bug being fixed.

**Comment by risdenk** [comment]
Committed d7d28a273c127d324787298ed253fae94b9f221d

**Comment by Functor10** [comment]
@risdenk The bug still exists. May I need to open another issue in JIRA?

**Comment by asereda-gs** [comment]
@Functor10  yes pls open new JIRA.

@michaelmior sorry I missed this PR and your message for some reason. I think among that gitbox noise.

**Pull request closed without merging**